tracks/elm/exercises/leap/mentoring.md

```diff
@@ -0,0 +1,90 @@
+This is a good exercise to discuss the describability of logic, as most students solve the problem using a conditional statement that is hard to describe in language, and there are other options.
+
+This exercise introduces `Basics.remainderBy`, the `let` / `in` syntax and partial application / currying.
+
+### Reasonable solutions
+
+The easiest / most common solution probably involves partially applying `Basics.remainderBy` and using a slightly complicated if statement:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+    let
+        divisibleBy number = 
+            Basics.remainderBy number year == 0         
+    in
+        (
+            divisibleBy 4 && 
+            not (divisibleBy 100)
+        )
+        || divisibleBy 400
```

Comment:
Would you not rather have

```elm
divisibleBy 4 && (not (divisibleBy 100) || divisibleBy 400)
```

to avoid checking for divisibility by 400 when you know that it isn't divisible by 4?

Reply:
Hi Simon, thanks for the review. This is a defendable point, I've made the change